### PR TITLE
[4.0] Added support for skipping pool creation while processing test data (ENT-4849)

### PR DESCRIPTION
**Pull request by Ceiu**
- Marketing products tagged with the "skip_pools" attribute will
  now be skipped during subscription/pool creation

### CODE REVIEW

**Review by JoshAlbrecht23 — APPROVED** [review]

**Comment by nikosmoum** [comment]
Jenkins is failing because of recent changes to the container image. I've backported the corresponding change in PR https://github.com/candlepin/candlepin/pull/3456. After that is merged, the build should pass.

**Comment by nikosmoum** [comment]
@Ceiu Looks like there was some weird issue with Jenkins' visibility of the candlepin-4.0-HOTFIX branch. It kept merging in to this change the commit right before the one that included the JDBC jar changes.
I added a commit (but then removed it) to this PR, which forced Jenkins to merge in the fresh state of the branch, so now it's working.

**Review by nikosmoum — APPROVED** [review]

**Pull request merged**
